packages/@aws-cdk/aws-dynamodb/lib/table.ts

```diff
@@ -642,56 +713,199 @@ abstract class TableBase extends Resource implements ITable {
+      operations.push(props.dimensions?.Operation);
+    }
+
+    if (props.operations) {
+      // user specified the operations property, use it.
+      // note this means that the Operation dimension is not passed because of the above validation.
+      operations.push(...props.operations);
+    }
+
+    const usingMetrics: Record<string, cloudwatch.IMetric> = {};
+
+    const denominator = [];
+    const numerator = [];
+
+    if (operations.length > 5 || operations.length === 0) {
+      // you must pass at least 1 operation since this is an operation dimensionality metric.
+      // you cannot pass more than 5 because this would make this metric unusable for alarms who have a 10 individual metric limit for math expressions.
+      // https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/AlarmThatSendsEmail.html#alarms-on-metric-math-expressions
+      throw new Error("Number of operations for the 'metricSuccessfulRequestLatency' metric should be between 1 and 5");
+    }
+
+    for (const operation of operations) {
+
+      // the name must start with a lowercase letter.
+      const operationName = operation.toLowerCase();
+
+      const valueMetricName = `${operationName}Value`;
+      usingMetrics[valueMetricName] = this.metric('SuccessfulRequestLatency', {
+        statistic: 'avg',
+        dimensions: {
+          TableName: this.tableName,
+          Operation: operation,
+        },
+        ...props,
+      });
+
+      const countMetricName = `${operationName}Count`;
+      usingMetrics[countMetricName] = this.metric('SuccessfulRequestLatency', {
+        statistic: 'n',
+        dimensions: {
+          TableName: this.tableName,
+          Operation: operation,
+        },
+        ...props,
+      });
+
+      denominator.push(countMetricName);
+      numerator.push(`${countMetricName} * ${valueMetricName}`);
+    }
+
+    return new cloudwatch.MathExpression({
+      expression: `(${numerator.join(' + ')}) / (${denominator.join(' + ')})`,
+      usingMetrics,
+      color: props?.color,
+      label: props?.label ?? 'Weighted average over all operations',
+      period: props?.period,
+    });
+
+  }
+
+  /**
+   * Metric for the successful request read latency of this table
    */
-  public metricSuccessfulRequestLatency(props?: cloudwatch.MetricOptions): cloudwatch.Metric {
-    return this.metric('SuccessfulRequestLatency', { statistic: 'avg', ...props });
+  public metricSuccessfulReadRequestLatency(props?: cloudwatch.MetricOptions): cloudwatch.IMetric {
+
+    if (props?.dimensions?.Operation) {
+      throw new Error("Operation dimension cannot be passed to the 'metricSuccessfulReadRequestLatency' method. Use 'metricSuccessfulRequestLatency' instead.");
+    }
+
+    return this.metricSuccessfulRequestLatency({
+      operations: [Operation.GET_ITEM, Operation.SCAN, Operation.QUERY, Operation.GET_RECORDS, Operation.BATCH_GET_ITEM],
```

Comment:
So we're just lucky that there are only 5 read operations, otherwise this wouldn't work? 

I do wonder how useful this metric is on the whole. There can be huge variations between a GET_ITEM and SCAN, and having that lumped into a single metric seems less useful. I wonder if dropping these new convenience methods (for now) would be pragmatic, and then have the community help us understand what combination(s) of these operations makes the most sense to be lumped together.

Reply:
![yes](https://media.giphy.com/media/3o6Zt7g9nH1nFGeBcQ/source.gif)